pkg/endpoints/credentials.go

```diff
@@ -343,3 +370,81 @@ func credentialToSecret(cred credential, namespace string, response *restful.Res
 	// Return secret
 	return &secret, true
 }
+
+// addSecretToSA - Add secret to service account 
+func (r Resource) addSecretToSA(saName string, secretName string , namespaceName string) bool {
+	sa := corev1.ServiceAccount{
+		Secrets: []corev1.ObjectReference {{Name: secretName},},
+	}
+
+	marshaledSA, err := json.Marshal(sa)
+	if err != nil {
+		logging.Log.Errorf("Failed to marshal payload")
+		return false
+	}
+
+	logging.Log.Debugf("Service account to patch is %s", saName)
+	logging.Log.Debugf("Namespace of service account is %s", namespaceName)
+	logging.Log.Debugf("Patch string %s", string(marshaledSA))
+
+	_, err = r.K8sClient.CoreV1().ServiceAccounts(namespaceName).
+		Get(saName, metav1.GetOptions{})
+
+	if err != nil {
+		logging.Log.Infof("Service Account %s doesn't exist. Creating..", saName)
+		_, err = r.K8sClient.CoreV1().ServiceAccounts(namespaceName).
+			Create(&corev1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: saName,},})
+		if err != nil {
+			logging.Log.Errorf("Error occurred: %s", err.Error())
+			return false
+		}
+	}
+
+	_, err = r.K8sClient.CoreV1().ServiceAccounts(namespaceName).
+		Patch(saName, typesPatch.StrategicMergePatchType, marshaledSA)
+
+	if err != nil {
+		logging.Log.Errorf("Error occurred: %s", err.Error())
+		return false
+	}
+	return true
+}
+
+// path - struct for patch operation
+type patch struct {
+	Op              string            `json:"op"`
```

Comment:
Is Op operation? Guessing so but may be unclear to a newcomer, suggesting Operation as field name